lib/src/assists/flutter/wrap_with_listenable_builder.dart

```diff
@@ -17,7 +18,16 @@ class WrapWithListenableBuilder extends DartAssist {
   ) async {
     if (!context.pubspec.isFlutterProject) return;
 
-    //TODO(Sam): Check for flutter version. ListenableBuilder is only available after version 3.10
+    // Its not neccessary to have flutter mentioned in the sdk block.
+    final flutterVersion = context.pubspec.environment?['flutter'];
+    final minimumRequiredVersion = Version(3, 10, 0);
+
+    switch (flutterVersion) {
+      case final Version version when version < minimumRequiredVersion:
+      case VersionRange(:final min)
+          when min != null && min < minimumRequiredVersion:
+        return;
+    }
```

Comment:
I have added the minimum version check but the problem is that there's no guarantee there will be flutter will be mentioned in the SDK block. There's no other way to get the version from the lock file. With the current implementation, we will show the assist if we are not sure about the flutter version. But we can also go the opposite way to only show the assist if we are definite about the flutter version. What do you think?

Reply:
I though about this as well. One way is to maintain a map of Flutter version to Dart version from https://docs.flutter.dev/release/archive?tab=linux.

For example, Flutter 3.10.0 uses Dart 3.0.0. We can check if there is flutter in `dependencies` and the min Dart SDK version in the `environment` instead.

However, this requires more work. We could also go with your approach.

Reply:
I think we can check for the dart version as you just said which btw is guaranteed to be there in the SDK block.  I'll update the code. What about that pub_semver package. Do we need to include it in the dependencies or just use it without specifying it in the dependencies for now?

Reply:
Yes. Add it to the pubspec if you need it. Otherwise you will get a lint warning.

Reply:
Yeah sure.